airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_stream.py

```diff
@@ -60,6 +60,14 @@ def primary_key(self, value: str) -> None:
         if not isinstance(value, property):
             self._primary_key = value
 
+    @property
+    def exit_on_rate_limit(self) -> bool:
+        return self.retriever.requester.exit_on_rate_limit  # type: ignore
```

Comment:
Am I correct in thinking that these mypy failures occur because DeclarativeStream's retriever is of the interface `Retriever` type, but the `exit_on_rate_limit` is only defined the `HttpRequester`?

If that's the case, I think we should not be ignoring this mypy check and instead be exposing the `exit_on_rate_limit` setter and getters on both the `Retriever` and `Requester` interfaces to address the error since now those methods will actually exist on the typed interfaces

Reply:
for this line mypy reports that `Retriever` has no attribute `requester`. So, I think we can just ignore it with a comment.